learning-bitmark/working-with-bitmarks/using-bitmark-shares.md

```diff
@@ -0,0 +1,631 @@
+  ```
+  ```json
+    {
+      "txId": "f61c41e7261508f9f65fa73c5af93a86e02ba82361e8f01afde9245d0bbe82dc",
+      "rawTx": "0100000001b6e8...846200000000"
+    }
+  ```
+
+<br>
+
+* Verify the grant shares transaction status
+
+  ```shell
+    $ bitmark-cli -n testing \
+      status -t \
+      fc88099d56c6f33bd90eb35937aa5d6e875c00017e7a78046ab3e5279b337e57
+  ```
+  ```json
+    {
+      "status": "Verified"
+    }
+
+    //Check again after some minutes
+    {
+      "status": "Confirmed"
+    }
+  ```
+
+<br>
+
+* Check the sender's balance
+
+  ```shell
+    $ bitmark-cli -n testing balance -o first
+  ```
+  ```json
+    {
+      "balances": [
+        {
+          "shareId": "b069f2956b828281dec040782eea3d63793ab4cf17c26f7639e95f6f3b20ba23",
+          "confirmed": 900,
+          "spend": 0,
+          "available": 900
+        }
+      ]
+    }
+  ```
+
+<br>
+
+* Check the receiver's balance
+
+  ```shell
+    $ bitmark-cli -n testing balance -o second
+  ```
+  ```json
+    {
+      "balances": [
+        {
+          "shareId": "0f366f3ee477a75c6304746f45296d804c08be06ad0941313371a17fe07d778f",
+          "confirmed": 2000,
+          "spend": 0,
+          "available": 2000
+        },
+        {
+          "shareId": "b069f2956b828281dec040782eea3d63793ab4cf17c26f7639e95f6f3b20ba23",
+          "confirmed": 100,
+          "spend": 0,
+          "available": 100
+        }
+      ]
+    }
+  ```
+
+<br>
+<br>
+
+## Swapping bitmark shares
+
+Two owners of different bitmark shares are able to swap their shares together. The swap requires signatures from both owners.
+
+A swap share transaction is a two signature transaction. Therefore, both the owners need to sign to transfer and accept the corresponding shares.
+
+  ```shell
+    # One of the owner initializes the swap shares transaction
+    $ bitmark-cli -n <network> -i <owner identity> \
+      swap -r <receiver> -s <shareId1> -q <quantity1> \
+      -S <shareId2> \
+      -Q <quantity2> \
+      -b <block_number>
+
+
+    # The other owner (the receiver) signs to swap the shares
+    $ bitmark-cli -n <network> -i <receiver identity> \
+      countersign -t <hex-data>
+  ```
+
+  >The `swap` command is to initialize a swap shares request, the request needs to be sign by the sender. Once the receiver signs to accept the request, the swap shares transaction is created and submitted to the blockchain. The `swap` command options:
+  >
+  >* `receiver` - The receiver's identity.
+  >
+  >* `shareId1` - The id of the sender's shares.
+  >
+  >* `quantity1` - The number of shares that the sender wants to swap with the receiver.
+  >
+  >* `shareId2` - The id of the receiver's shares for which the sender wants to swap.
+  >
+  >* `quantity2` - The number of the receiver's shares for which the sender wants to swap.
+  >
+  >* `block_number` - This provides a time-limit to let the request expire and become invalid. Every 30 blocks from the current block adds one hour of expiration.
```

Comment:
You probably need to explain how to get the current block number. As explained, people are likely to write something like "-b 30" if they want it to be 30 blocks on, rather than "-b (current + 30)".

Reply:
Updated.

Reply:
@hxw : I think Shannon's comment makes sense. Can we update the CLI to do so? (`-b 30` instead of `-b <current block height + 30>`)